### PR TITLE
Added BFD rules

### PR DESCRIPTION
**Pull request by gvarao**

### CODE REVIEW

File: juniper_official/routing/bfd-session-state.rule [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            keys session-neighbor;
+            synopsis "bfd session state analyzer";
+            description "This rule collects bfd session state periodically and notifies in case of anomalies";
+            sensor bfd-sensor {
+                synopsis "bfd netconf based iAgent sensor definition";
+                description "iAgent sensor defined in yml file using netconf rpc/cli command and defined fields using pyez tables&views to collect data from network device";
+                iAgent {
+                    file bfd-session-state.yml;
+                    table BfdSessionTable;
+                    frequency 60s;
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: 3min

File: juniper_official/routing/bfd-session-state.rule [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    where "session-neighbor =~ /{{bfd-session-id}}/";
+                    path session-neighbor;
+                }
+                type string;
+                description "BFD neighbor ip";
+            }
+            trigger bfd-session-state {
+                synopsis "BFD neighbor state kpi";
+                description "Sets health based on BFD neighbor state.";
+                frequency 1.5offset;
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: 1 offset

File: juniper_official/routing/bfd-session-state.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        detail: True
+    item: //bfd-session
+    key: session-neighbor
+    view: BfdSessionView
+
+BfdSessionView:
+    fields:
+        session-state: session-state
+        session-interface: session-interface
+        remote-state: remote-state
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.yml:14]: scale? and xslt

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/routing/bfd-session-state.rule (hunk 1 above)
Comment (gvarao) [reply juniper_official/routing/bfd-session-state.rule]: Increased to 180Sec

File: juniper_official/routing/bfd-session-state.rule (hunk 2 above)
Comment (gvarao) [reply juniper_official/routing/bfd-session-state.rule]: Modified

File: juniper_official/routing/bfd-session-state.yml (hunk 3 above)
Comment (gvarao) [reply juniper_official/routing/bfd-session-state.yml:14]: It will be addressed in FH-2810

File: juniper_official/routing/bfd-session-state.rule [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                version 2;
+                contributor juniper;
+                category advanced;
+                is-scaling-rule {
+                    description "Fields:session-neighbor ; Directly impacted by number of bfd sessions running in each network device";
+                }				
+                supported-healthbot-version 1.0.1;
+                supported-devices {
+                    juniper {
+                        operating-system junos {
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: What about EVO and and ACX?

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/routing/bfd-session-state.rule (hunk 4 above)
Comment (gvarao) [reply juniper_official/routing/bfd-session-state.rule]: Added EVO and ACX in meta data

File: juniper_official/routing/bfd-session-state.rule [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                            }
+                            products QFX {
+                                platforms All {
+                                    releases 15.2R1 {
+                                        release-support min-supported-release;
+                                    }
+                                }							
+                            }
+                        }
+                        operating-system junosEvolved {
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: there should be one for vmx as well

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/routing/bfd-session-state.rule [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                category advanced;
+                is-scaling-rule {
+                    description "Fields:session-neighbor ; Directly impacted by number of bfd sessions running in each network device";
+                }				
+                supported-healthbot-version 1.0.1;
+                supported-devices {
+                    juniper {
+                        operating-system junos {
+                            products EX {
+                                platforms All {
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: there is nothing like platforms ALL, this was a mistake in older rules, either mention each of the products or put a default sensor at platform level

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/routing/bfd-session-state.rule (hunk 5 above)
Comment (gvarao) [reply juniper_official/routing/bfd-session-state.rule]: VMX will be added after qualification

**Review by gaddevijay — APPROVED** [review]

**Pull request merged**